pkg/getter/httpgetter.go

```diff
@@ -123,21 +127,21 @@ func (g *HTTPGetter) httpClient() (*http.Client, error) {
 		}
 		tlsConf.ServerName = sni
 
-		transport.TLSClientConfig = tlsConf
+		g.transport.TLSClientConfig = tlsConf
```

Comment:
For users providing a non-ephemeral transport would it make sense for the `httpgetter` to amend its internals? I had the same debate whilst looking at this problem, and feels problematic the possibility of passing a transport to this func and have side effects in other places that share the same transport instance.

For example, depending on the order of the calls, something else completely unrelated to helm can be called after this func and from that point onwards have `InsecureSkipVerify` set to `true`, leaving callers in a insecure position.

To avoid such side effects, it may be worth making the caller fully responsible for configuring the transport if they so wish. So they either rely on `httpgetter` to manage the transport, or they take that onto themselves. Using an in-between approach may lead to unexpected and potentially less secure outcomes.